### PR TITLE
Added masterless support to utils/minions.py:_pki_minions function

### PR DESCRIPTION
**Pull request by mcd1992**
### What does this PR do?
Adds support to utils/minions.py's _pki_minions function for masterless calls

### What issues does this PR fix or reference?
No known issues/pr open related to this but I ran into this being an issue when using ext_pillar & file_tree on a masterless dev. environment. The file_tree module would use the minions utility library to get all minions / nodegroups from PKI files which don't exist on the masterless minion.

### Previous Behavior
```
$ salt-call --local pillar.get nodegroups 
[ERROR   ] Failed matching available minions with compound pattern: *
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/salt/utils/minions.py", line 684, in check_minions
    _res = check_func(expr, delimiter, greedy)
  File "/usr/lib/python2.7/site-packages/salt/utils/minions.py", line 476, in _check_compound_minions
    minions = set(self._pki_minions())
  File "/usr/lib/python2.7/site-packages/salt/utils/minions.py", line 250, in _pki_minions
    if self.opts['key_cache'] and os.path.exists(pki_cache_fn):
KeyError: u'key_cache'
```

### Tests written?
No

### Commits signed with GPG?
Yes

### CODE REVIEW

**Comment by mcd1992** [comment]
Made changes into single logic statement as requested in https://github.com/saltstack/salt/pull/47340

Hopefully this 'style' for multi-line is acceptable, didn't see many examples in existing code to base it off of.

**Comment by mcd1992** [comment]
Man I suck at git. Forgot to GPG sign and github doesn't like my git amend and force...

**Comment by mcd1992** [comment]
Aaand the fast forward is showing up in here.... The git history itself seems clean but the github PR is associating the other commits with this PR? Not sure if  this will cause an issue other than making the PR look bloated, let me know if I should just open another PR. If so I can base it off the 2018.3 branch anyway since this is more of a bugfix than a feature. Didn't realize the difference between develop branch and the others when I first forked this.

**Comment by rallytime** [comment]
@mcd1992 We have some docs about that here, specifically step #6 about rebasing, if that helps?

https://docs.saltstack.com/en/latest/topics/development/contributing.html#sending-a-github-pull-request

**Comment by mcd1992** [comment]
Ah gotcha, thanks rally. Guess I still need to read up on the differences between a rebase and fast-forward.

**Review by cachedout — APPROVED** [review]

**Comment by rallytime** [comment]
re-run py

File: salt/utils/minions.py [hunk 1]
```diff
@@ -239,9 +239,11 @@ def _pki_minions(self):
         Retreive complete minion list from PKI dir.
         Respects cache if configured
         '''
-        if self.opts.get('__role') == 'master' and self.opts.get('__cli') == 'salt-run':
-            # Compiling pillar directly on the master, just return the master's
-            # ID as that is the only one that is available.
+        opts_role = self.opts.get('__role')
+        if ((opts_role == 'master' and self.opts.get('__cli') == 'salt-run') or
+         (opts_role == 'minion')):
```
Comment (isbm) [inline salt/utils/minions.py]: I think we have an issues with the indent here. Maybe just put it in one line, it anyway reads better then.

**Review by isbm — CHANGES_REQUESTED** [review]

**Comment by rallytime** [comment]
@mcd1992 This has a lint error: https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/21716/violations/file/salt/utils/minions.py/

File: salt/utils/minions.py [hunk 2]
```diff
@@ -240,8 +240,7 @@ def _pki_minions(self):
         Respects cache if configured
         '''
         opts_role = self.opts.get('__role')
-        if ((opts_role == 'master' and self.opts.get('__cli') == 'salt-run') or
-         (opts_role == 'minion')):
+        if ((opts_role == 'master' and self.opts.get('__cli') == 'salt-run') or (opts_role == 'minion')):
```
Comment (isbm) [inline salt/utils/minions.py]: Better, but too much parenthesis here :wink: Lint already said that too, BTW.

**Review by isbm — APPROVED** [review]

**Review by isbm — CHANGES_REQUESTED** [review]
Oops, clicked wrong selector. Please remove parenthesis and so fix Lint. :-)

**Comment by mcd1992** [comment]
I was unable to get the C0325 pylint error to show up locally but its a simple enough fix anyway.

https://gist.github.com/mcd1992/ba02c7978e9dcd69c2f29885d96ad6cd

**Pull request merged**